src/main/java/seedu/address/model/Model.java

```diff
@@ -19,14 +20,31 @@
     void resetData(ReadOnlyTaskBook newData);
 
     /** Returns the TaskBook */
-    ReadOnlyTaskBook getAddressBook();
+    ReadOnlyTaskBook getTaskBook();
 
     /** Deletes the given task. */
     void deleteTask(Task target) throws TaskNotFoundException;
 
     /** Adds the given task */
     void addTask(Task task);
 
+    //undo and redo
+    /** Undo prev action that modifies data**/
+    Command undo();
+
+    /** reset stack of redoable actions when a non undo modifying data command is called**/
+    void resetRedoables();
+
+    void recordState(Command command);
+
+    boolean hasUncommittedChanges();
+
+    void discardRecentCommit();
```

Comment:
So, I don't think this API is well designed because it is not clear to me what methods I need to call in order to integrate with the undo/redo system. Do I call recordState() before or after I made the changes? When do I call resetRedoables()? What happens if I do not call it? How about discardRecentCommit()?

Adding comments might help a bit, but it still places a big burden on the API users to ensure that the API is used in a way that does not break the application.